src/Module/Login.php

```diff
@@ -186,7 +190,13 @@ public static function sessionAuth()
 					]
 				);
 				if (DBA::isResult($user)) {
-					if ($data->hash != Authentication::getCookieHashForUser($user)) {
+					// Time safe comparision of the two hashes.
+					$validSession = hash_equals(
```

Comment:
Thanks for this improvement, but if `$validSession` isn't used later in the script, please move the `hash_equals()` call inside the `if` condition.